accessibility updates to portals section (#200)
ubccr/ccrdocs: pages/portals/ood.md

Under the `Files` tab all users will see the `Home Directory` option.  On the systems this points to `/user/[CCRusername]`.  If you have access to a shared project or global scratch directory, you will see this in the Files drop down menu. You can verify what you should have access to by viewing your [active allocations in ColdFront](coldfront.md) (https://coldfront.ccr.buffalo.edu)  

> You can add links to OneDrive and UB Box in the Files app as well.  You will need to use the command line to setup authentication with your UB Box and/or OneDrive account(s).  Refer to the documentation(../hpc/data-transfer.md#rclone) about using `rclone` to setup the connections with UB Box and/or OneDrive.  Once this authentication is setup, you'll see the links in the OnDemand Files app.  If you've done this setup after logging into OnDemand, you may need to restart the OnDemand web service to see the changes.  To do so, under the `Help` menu click on `Restart Web Server`  

Reviewer: ```suggestion
You can add links to OneDrive and UB Box in the Files app as well.  You will need to use the command line to setup authentication with your UB Box and/or OneDrive account(s).  Refer to the [`rclone` documentation](../hpc/data-transfer.md#rclone) to setup the connections with UB Box and/or OneDrive.  Once this authentication is setup, you'll see the links in the OnDemand Files app.  If you've done this setup after logging into OnDemand, you may need to restart the OnDemand web service to see the changes.  To do so, under the `Help` menu click on `Restart Web Server`  
```

#### Desktop Interactive Apps   

> The interactive desktop app launches a job on a compute node in the UB-HPC or Faculty cluster and provides the user with a Linux desktop experience.  The interactive desktop app form requires the user to select which cluster they want the job submitted to, as well as which partition, qos, and account to use for the job.  Though users will see all partitions listed in the Faculty cluster, they will only be able to use the one(s) that match the QOS values shown in their drop down menu.  For example, to use the `ub-laser` partition in the faculty cluster, you must select the `ub-laser` QOS.  Access to partitions and QOS values is managed through [allocations in ColdFront](coldfront.md#request-an-allocation).  Interactive desktop sessions are limited to one node; however if you have other requirements such as number of cores (CPUs), GPUs, amount of memory (RAM), run time (walltime), or specific node features you may enter that in the appropriate form fields.  If you do not care what type of hardware your desktop runs on, you can leave these blank and the scheduler will allocate one CPU and 2.8GB of RAM for your session.  The [default walltime](../hpc/jobs.md#slurm-directives-partitions--qos) varies by partition.

Reviewer: ```suggestion
The interactive desktop app launches a job on a compute node in the UB-HPC or Faculty cluster and provides the user with a Linux desktop experience.  The interactive desktop app form requires the user to select which cluster they want the job submitted to, as well as which partition, qos, and account to use for the job.  Though users will see all partitions listed in the Faculty cluster, they will only be able to use the one(s) that match the QOS values shown in their drop down menu.  For example, to use the `ub-laser` partition in the faculty cluster, you must select the `ub-laser` QOS.  Access to partitions and QOS values is managed through [allocations in ColdFront](coldfront.md#request-an-allocation).  Interactive desktop sessions are limited to one node; however if you have other requirements such as number of cores (CPUs), GPUs, amount of memory (RAM), run time (walltime), or specific node features you may enter that in the appropriate form fields.  If you do not care what type of hardware your desktop runs on, you can leave these blank and the scheduler will allocate one CPU and 2.8GB of RAM for your session.  The [default walltime](../hpc/jobs.md#slurm-directives-partitions-qos) varies by partition.
```

**None of the OnDemand applications work with the `arm64` partition. Users must use [interactive or batch jobs](../hpc/jobs.md#running-jobs) for these nodes.**

#### Matlab App  

Matlab is available to UB faculty, staff, and students through the university's site license. Matlab is not available to industry customers.  It can be run on the command line via a [batch job](../hpc/jobs.md#running-applications-on-the-clusters), launching the GUI from an [interactive desktop](#desktop-interactive-apps) session, or using the Matlab interactive app in OnDemand.  When using the interactive app, select which version of Matlab you'd like to run from the drop down menu.  When the job launches on a node, the Matlab GUI will display automatically.  

#### RStudio App

> The RStudio interactive app in OnDemand allows the user to select where or how to launch RStudio.  Users can select from a drop down menu that includes CCR's default R installation or containerized options.  CCR provides 4 additional versions of R, RStudio, and Bioconductor via containers from [Bioconductor](https://hub.docker.com/u/bioconductor). Alternatively, users may choose to utilize their own container for launching RStudio.  You may download alternate versions from Bioconductor or create your own.  As long as the container includes the RStudio application, it should launch the GUI properly when using this app.  When using your own container image, please enter the full path and name of the Apptainer image you'd like to use.  Refer to CCR's [container documentation](../howto/containerization.md) for more information.  

Reviewer: ```suggestion
The RStudio interactive app in OnDemand allows the user to select where or how to launch RStudio.  Users can select from a drop down menu that includes CCR's default R installation or containerized options.  CCR provides 4 additional versions of R, RStudio, and Bioconductor via containers from [Bioconductor](https://hub.docker.com/u/bioconductor). Alternatively, users may choose to utilize their own container for launching RStudio.  You may download alternate versions from Bioconductor or create your own.  As long as the container includes the RStudio application, it should launch the GUI properly when using this app.  When using your own container image, please enter the full path and name of the Apptainer image you'd like to use.  Refer to [CCR's container documentation](../howto/containerization.md) for more information.  
```

These are just a few of the oddities to note about how VSCode functions at CCR:

  - Loading modules when VSCode starts does not work the same way as described above for the Jupyter Lab/Notebook app.  You will need to load the software modules in a VSCode terminal after the application has started.  Despite doing this, not all software modules will work correctly in the VSCode terminal.
  - VSCode can't detect the CCR software repository and load Python and its dependencies. When choosing an interpreter, the default Python installation will be the system Python and should NOT be used. Instead, you must use a properly configured [virtual environment](../howto/python.md#virtual-environments).

>   - If creating a virtual environment in VSCode, use the terminal and [these instructions](../howto/python.md#virtual-environments), NOT the Python extension and its options for creating virtual environments.  

Reviewer: ```suggestion
  - If creating a virtual environment in VSCode, use the terminal, follow [CCR's virtual environment documentation](../howto/python.md#virtual-environments), NOT the Python extension and its options for creating virtual environments.  
```

### My Interactive Sessions  

The interactive sessions section of the OnDemand portal displays your currently running sessions and those that just recently finished.  The completed jobs stay in this list for 2 days allowing you to access the link to the jobs' output files.  You can remove the links by clicking the `Delete` button.  However, this does not delete the [job data directory and output files](../howto/ood-trouble.md#where-can-i-find-the-output-files-for-my-ondemand-sessions).

!!! Note "Managing your OnDemand Disk Usage"
    To clear up disk space in your home directory, you can remove directories for completed jobs when you're done retrieving the job output data.  You can find these in subdirectories under: `/user/[CCRusername]/ondemand/data/sys/dashboard/batch_connect/sys`  

### Develop (Sandbox)  

> OnDemand allows users the ability to create their own interactive apps and run them in a 'sandbox' environment.  CCR does not turn this on by default, however.  Please [contact CCR Help](../help.md) to request this feature be enabled for your account.  Once it is, you'll find the `My Sandbox Apps (Development)` option under the `</>Develop` menu in OnDemand.  Please refer to the documentation on app development provided by the [Open OnDemand developers here](https://osc.github.io/ood-documentation/latest/app-development).  

Reviewer: ```suggestion
OnDemand allows users the ability to create their own interactive apps and run them in a 'sandbox' environment.  Please note, CCR does NOT enable this feature by default; please request this feature be enabled for your account via [CCR Help](../help.md).  Once it is enabled, you'll find the `My Sandbox Apps (Development)` option under the `</>Develop` menu in OnDemand.  Please refer to the documentation on app development provided by the [Open OnDemand developers](https://osc.github.io/ood-documentation/latest/app-development) for more information.  
```

### Help  

The Help menu offers links to CCR's support contact information, documentation, and the ability to restart your OnDemand web process.  Occasionally, you'll find your OnDemand session may be slow or unresponsive.  Users can choose the `Restart Web Server` option to restart the process that runs their individual OnDemand session.  Most of the time you will not need to do this unless directed to by CCR staff.  You may use the `Submit Support Ticket` option to easily submit a ticket to CCR help.  If you're contacting us regarding a specific job, please use the `Submit Support Ticket` button on the job card so that all the relevent job information is included in the ticket.  Then we don't have to ask you for it!    

### Troubleshooting  

> Please refer to CCR's OnDemand [troubleshooting guide](../howto/ood-trouble.md).

Reviewer: ```suggestion
Please refer to [CCR's OnDemand troubleshooting guide](../howto/ood-trouble.md).
```
